platform/android/MapboxGLAndroidSDK/src/test/java/org/maplibre/android/location/engine/MapLibreFusedLocationEngineImplTest.kt

```diff
@@ -0,0 +1,164 @@
+    }
+
+    @Test
+    fun requestLocationUpdatesIndoors() {
+        val request = LocationEngineRequest.Builder(10)
+                .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY).build()
+        // J2K: IDE suggestion "as LocationEngineCallback<LocationEngineResult>"
+        val callback: LocationEngineCallback<LocationEngineResult> = Mockito.mock(LocationEngineCallback::class.java) as LocationEngineCallback<LocationEngineResult>
+        val looper = Mockito.mock(Looper::class.java)
+        Mockito.`when`(locationManagerMock!!.getBestProvider(ArgumentMatchers.any(Criteria::class.java), ArgumentMatchers.anyBoolean())).thenReturn("network")
+        engine!!.requestLocationUpdates(request, callback, looper)
+        Mockito.verify(locationManagerMock, Mockito.times(1)).requestLocationUpdates(ArgumentMatchers.anyString(),
+                ArgumentMatchers.anyLong(), ArgumentMatchers.anyFloat(), ArgumentMatchers.any(LocationListener::class.java), ArgumentMatchers.any(Looper::class.java))
+    }
+
+    @Test
+    fun requestLocationUpdatesOutdoorsWithPendingIntent() {
+        val request = LocationEngineRequest.Builder(10)
+                .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY).build()
+        val pendingIntent = Mockito.mock(PendingIntent::class.java)
+        Mockito.`when`(locationManagerMock!!.getBestProvider(ArgumentMatchers.any(Criteria::class.java), ArgumentMatchers.anyBoolean())).thenReturn("gps")
+        engine!!.requestLocationUpdates(request, pendingIntent)
+        Mockito.verify(locationManagerMock, Mockito.times(2)).requestLocationUpdates(ArgumentMatchers.anyString(),
+                ArgumentMatchers.anyLong(), ArgumentMatchers.anyFloat(), ArgumentMatchers.any(PendingIntent::class.java))
+    }
+
+    @Test
+    fun requestLocationUpdatesIndoorsWithPendingIntent() {
+        val request = LocationEngineRequest.Builder(10)
+                .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY).build()
+        val pendingIntent = Mockito.mock(PendingIntent::class.java)
+        Mockito.`when`(locationManagerMock!!.getBestProvider(ArgumentMatchers.any(Criteria::class.java), ArgumentMatchers.anyBoolean())).thenReturn("network")
+        engine!!.requestLocationUpdates(request, pendingIntent)
+        Mockito.verify(locationManagerMock, Mockito.times(1)).requestLocationUpdates(ArgumentMatchers.anyString(),
+                ArgumentMatchers.anyLong(), ArgumentMatchers.anyFloat(), ArgumentMatchers.any(PendingIntent::class.java))
+    }
+
+    // J2K: rewrite test to not use "@get", rename to "getLastLocationNull"
+    @Test(expected = NullPointerException::class)
+    fun getLastLocationNull() {
+        engine!!.getLastLocation(null!!)
+    }
+
+    // J2K: changed "null" to "null!!"
+    @Test(expected = NullPointerException::class)
+    fun requestLocationUpdatesNullCallback() {
+        engine!!.requestLocationUpdates(null!!, null!!, null!!)
```

Comment:
Using a non-null assertion operator on `null` will throw a null pointer exception, but it won't even reach into the method, so you need a different way to pass `null` here.

Reply:
I spend the most time here to lookup for a solution, did not find something specific.

Reply:
OK I suggest we remove this test then, because the NullPointerException does not come from requestLocationUpdates right now.

Reply:
disabled them and added a TODO + Issue so this gets not forgotten. I have the feeling that if a library is used from java, then the null-ptr tests need to be in Java - but I'm not sure about it. 
98e4632